src/System.Net.Security/src/System/Net/SslStreamPal.Windows.cs

```diff
@@ -126,8 +127,13 @@ public static SafeFreeCredentials AcquireCredentialsHandle(X509Certificate certi
             return AcquireCredentialsHandle(direction, secureCredential);
         }
 
-        public static SecurityStatusPal EncryptMessage(SafeDeleteContext securityContext, byte[] writeBuffer, int size, int headerSize, int trailerSize, out int resultSize)
+        public static SecurityStatusPal EncryptMessage(SafeDeleteContext securityContext, byte[] input, int offset, int size, int headerSize, int trailerSize, ref byte[] output, out int resultSize)
         {
+            byte[] writeBuffer = output;
+
+            // Copy the input into the output buffer to prepare for SCHANNEL's expectations
+            Buffer.BlockCopy(input, offset, writeBuffer, headerSize, size);
+
```

Comment:
That's difficult given the surface area of SslStream.  The different ciphers and TLS protocol levels each produce different values for the header and trailer.  The test that would have caught this, System.Net.Security.Tests.SslStreamStreamToStreamTest.SslStream_StreamToStream_LargeWrites_Sync_Success ends up using AES256-GCM-SHA384. which has less framing overhead than ECDHE-RSA-AES256-SHA384 (the cipher mode that caught the error).

The big write was VirtualNetworkStream.Write(16384, 0, 16340); meaning that for AES256-GCM-SHA384 the overhead calculation was 44 bytes larger than OpenSsl actually needed.